shorthand_connect/tags/1.3.18/includes/api-v2.php

```diff
@@ -0,0 +1,192 @@
+	return json_decode($body);
+}
+
+function sh_get_profile() {
+	$tokeninfo = array();
+
+	$data = sh_v2_api_get_json('/v2/token-info', array());
+	if ($data && isset($data->organisation_id)) {
+		$tokeninfo['username'] = $data->name . ' ('.$data->token_type.' Token)';
+		$tokeninfo['gravatar'] = $data->logo;
+		$tokeninfo = (object)$tokeninfo;
+	}
+
+	return $tokeninfo;
+}
+
+function sh_get_stories() {
+	$stories = null;
+
+	$data = sh_v2_api_get_json('/v2/stories', array('timeout' => '240'));
+	if($data) {
+		$stories = array();
+		//Something went wrong
+		if (isset($data->status) && $data->status) {
+			return null;
+		}
+		foreach($data as $storydata) {
+			$story = array(
+				'image' => $storydata->signedCover,
+				'id' => $storydata->id,
+				'metadata' => (object)array(
+					'description' => $storydata->description
+				),
+				'title' => $storydata->title,
+				'story_version' => ''.$storydata->version
+			);
+			$stories[] = (object)$story;
+		}
+	}
+	
+	return $stories;
+}
+
+function sh_get_story_path($post_id, $story_id) {
+	WP_Filesystem();
+	global $wp_filesystem;
+	if ( ! is_a( $wp_filesystem, 'WP_Filesystem_Base') ) {
+		$creds = request_filesystem_credentials( site_url() );
+		wp_filesystem( $creds );
+	}
+	$destination = wp_upload_dir();
+	$destination_path = $destination['path'].'/shorthand/'.$post_id.'/'.$story_id;
+	if(!file_exists($destination_path)) {
+		$destination_path = null;
+	}
+	return $destination_path;
+}
+
+function sh_get_story_url($post_id, $story_id) {
+	WP_Filesystem();
+	global $wp_filesystem;
+
+	if ( ! is_a( $wp_filesystem, 'WP_Filesystem_Base') ) {
+		$creds = request_filesystem_credentials( site_url() );
+		wp_filesystem( $creds );
+	}
+	$destination = wp_upload_dir();
+	$destination_url = $destination['url'].'/shorthand/'.$post_id.'/'.$story_id;
+	return $destination_url;
+}
+
+function sh_copy_story($post_id, $story_id) {
+
+	wp_raise_memory_limit('admin');
+	WP_Filesystem();
+
+	global $wp_filesystem;
+
+	if ( ! is_a( $wp_filesystem, 'WP_Filesystem_Base') ) {
+		$creds = request_filesystem_credentials( site_url() );
+		wp_filesystem( $creds );
+	}
+
+	$destination = wp_upload_dir();
+	$tmpdir = get_temp_dir();
+	$destination_path = $destination['path'].'/shorthand/'.$post_id;
+
+	$story = array();
+
+	//Attempt to connect to the server
+	$zipfile = wp_tempnam('sh_zip',$tmpdir);
+	$unzipdir = wp_tempnam('sh_unzip',$tmpdir).'_dir';
+	$response = sh_v2_api_get('/v2/stories/'.$story_id, array(
+		'timeout' => '600',
+		'stream' => true,
+		'filename' => $zipfile
+	));
+	if (is_wp_error($response)) {
+		$story['error'] = array(
+			'pretty' => ' Request to Shorthand failed',
+			'error' => $response->get_error_message($response)
+		);
+	} else if (!$response || $response['response']['code'] != 200) {
+		$story['error'] = array(
+			'pretty' => 'Request to Shorthand failed; check the token is configured correctly',
+			'response' => $response
+		);
+	} else {
+		if(defined('WPCOM_IS_VIP_ENV')){
```

Comment:
Both the `if` and `else` are mostly the same. Even if they can't both use the functions used by the VIP section, most of the handling code _is_ the same.